src/Client/Provider/GithubClient.php

```diff
@@ -22,7 +22,29 @@ class GithubClient extends OAuth2Client
      */
     public function fetchUserFromToken(AccessToken $accessToken)
     {
-        return parent::fetchUserFromToken($accessToken);
+        $resourceOwner = parent::fetchUserFromToken($accessToken);
+
+        if (null === $resourceOwner->getEmail()) {
```

Comment:
We should invert the logic so w can return immediately in the “normal” case - it’s just avoids nesting in the if.

Reply:
I don't think it will be better because the ressource owner here is returned only if a primary email is found on the foreach.

So I don't have default return if I reverse this condition.

Reply:
Scope `user:email` is needed as far as I now